providers/evercrypt/fst/EverCrypt.Hash.fst

```diff
@@ -410,8 +410,6 @@ let update_multi #a s prevlen blocks len =
 
 #pop-options
 
-// Re-using the higher-order stateful combinator to get an instance of
-// update_last that is capable of calling Vale under the hood
 let update_last_256 s prev_len input input_len =
```

Comment:
can we get rid of the definition entirely? or is there a reason (e.g. different `val`)?

Reply:
We might be able to, however, it is used in other places in EverCrypt (HMAC, Incremental). So far, I kept it to avoid a larger diff, but we could remove it altogether, and call directly Hacl.Hash.SHA2.update_last in these modules.
We could also remove the wrapper for update_last_224 since I don't think it brings much value now that we are not using the generic mk_update_last combinator anymore.
Wdyt?

Reply:
as long as Hacl.Hash.SHA2.update_last_224 is an alias for Hacl.Hash.SHA2.update_last_256 (or vice-versa), we're good -- the goal is to avoid code duplication, both in C and in the Low* source

Reply:
This is going through several layers of definitions, but it seems all of them are inlined to in the end yield `Hacl.SHA2.Scalar32.update_last_224`, which is itself defined as `Hacl.SHA2.Scalar32.update_last_256` with a proof of equivalence. I'll make the change.